src/extension/wps/src/test/java/au/org/emii/gogoduck/download/ParallelDownloadManagerTest.java

```diff
@@ -0,0 +1,235 @@
+package au.org.emii.gogoduck.download;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.junit.Assert.assertEquals;
+
+public class ParallelDownloadManagerTest {
+
+    private static final long KiB = 1024L;
+    private static final int MILLISECONDS_IN_A_SECOND = 1000;
+
+    private Path downloadDir;
+    private Downloader downloader;
+    private ExecutorService pool;
+
+    @Before
+    public void setup() throws IOException {
+        downloadDir = Files.createTempDirectory("download.");
+        downloader = new Downloader(60*MILLISECONDS_IN_A_SECOND, 60*MILLISECONDS_IN_A_SECOND);
+        pool = Executors.newFixedThreadPool(8);
+    }
```

Comment:
Maybe we can move this to config and access it via au.org.emii.core.Config. It is fine here for testing purposes.